client/signup/config/flows.js

```diff
@@ -238,11 +239,72 @@ const Flows = {
 			flow = filterDesignTypeInFlow( flow );
 		}
 
-		return flow;
+		return Flows.getABTestFilteredFlow( flowName, flow );
 	},
 
 	getFlows() {
 		return flows;
+	},
+
+	/**
+	 * Preload AB Test variations after a certain step has been completed.
+	 *
+	 * This gives the option to set the AB variation as late as possible in the
+	 * signup flow.
+	 *
+	 * Currently only the `main` flow is whitelisted.
+	 *
+	 * @param {String} flowName The current flow
+	 * @param {String} stepName The step that is being completed right now
+	 */
+	preloadABTestVariationsForStep( flowName, stepName ) {
```

Comment:
I deliberately used `getABTestVariation` like it is here, because `abtest` sets the variation when it is called.

`getFlow` is the single point of truth for Signup, that gives the current flow configuration. It is called before every step and at the start of Signup.

For testing new steps, we don't want to set the variation until the last possible moment, which is right before the step (in this case - when the previous one ends). If we set the variation in the beginning it may skew the test with all the people that will drop off Signup even before reaching the step. This will require us to run the test longer, so we can gather  more accurate data. If we set the variation right before the step, the only users that will get it will be users that reach the step and it will more accurately measure the step success and attrition rates. 

That's why when `getFlow` gets called, it modifies the flow to what has been already set (e.g. a user resuming signup after page reload or if they navigated away from signup for some reason), so their progress will be properly loaded. Imagine their confusion if they open up signup and some step they already filled is missing and doesn't get sent to the API.

With this, the `preloadABTestVariationsForStep` method updates the flow exactly when it is needed, i.e. when the step is reached. 

cc @meremagee